packages/nextjs/pages/index.tsx

```diff
@@ -1,59 +1,204 @@
+import { useCallback, useEffect, useState } from "react";
+import Image from "next/image";
 import Link from "next/link";
+import { useRouter } from "next/router";
 import type { NextPage } from "next";
-import { BugAntIcon, MagnifyingGlassIcon } from "@heroicons/react/24/outline";
 import { MetaHeader } from "~~/components/MetaHeader";
+import { MiniFooter } from "~~/components/MiniFooter";
+import { AddressInput, InputBase } from "~~/components/scaffold-eth";
 
 const Home: NextPage = () => {
+  const [activeTab, setActiveTab] = useState("verifiedContract");
+  const [network, setNetwork] = useState("mainnet");
+  const [verifiedContractAddress, setVerifiedContractAddress] = useState("");
+  const [abiContractAddress, setAbiContractAddress] = useState("");
+  const [contractAbi, setContractAbi] = useState("");
+
+  const [isAbiAvailable, setIsAbiAvailable] = useState(false);
+
+  const tabNames = ["verifiedContract", "addressAbi"];
+
+  const getTransitionClasses = (tabName: any) => {
+    const baseClasses = "absolute inset-0 w-full transition-transform duration-300 ease-in-out px-1";
+    let translateClasses = "";
+
+    const currentTabIndex = tabNames.indexOf(activeTab);
+    const targetTabIndex = tabNames.indexOf(tabName);
+
+    if (currentTabIndex === targetTabIndex) {
+      translateClasses = "translate-x-0";
+    } else if (currentTabIndex < targetTabIndex) {
+      translateClasses = activeTab ? "translate-x-full" : "-translate-x-full";
+    } else {
+      translateClasses = activeTab ? "-translate-x-full" : "translate-x-full";
+    }
+
+    return `${baseClasses} ${translateClasses}`;
+  };
+
+  const router = useRouter();
+
+  const fetchContractABI = async () => {
+    const apiKey = process.env.NEXT_PUBLIC_ETHERSCAN_API_KEY;
+    const url = `https://api.etherscan.io/api?module=contract&action=getabi&address=${verifiedContractAddress}&apikey=${apiKey}`;
+
+    try {
+      const response = await fetch(url);
+      const data = await response.json();
+      if (data.status === "1") {
+        console.log("Contract ABI:", data.result);
+        setIsAbiAvailable(true);
+      } else {
+        console.error("Error fetching ABI:", data.result);
+        setIsAbiAvailable(false);
+      }
+    } catch (error) {
+      console.error("Error fetching ABI:", error);
+      setIsAbiAvailable(false);
+    }
+  };
+
+  useEffect(() => {
+    if (verifiedContractAddress) {
```

Comment:
Lets use `verifiedContractAddress && isAddress(verifiedContractAddress)` check here, because other wise we are hitting etherscan api on every stroke